torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -546,11 +546,15 @@ static bool is_int_list(PyObject* obj, int broadcast_size) {
     }
     auto item = py::reinterpret_steal<py::object>(
         PySequence_GetItem(obj, 0));
+    if (THPUtils_checkIndex(item.ptr())) {
+      return true;
+    }
+    // NOTE: JIT tracer allows arbitrary scalar tensors to act as ints
+    // in an intlist argument. Even float or complex scalar tensors.
     return (
-      THPUtils_checkIndex(item.ptr()) ||
-      // NOTE: JIT tracer allows arbitrary scalar tensors to act as ints
-      // in an intlist argument. Even float or complex scalar tensors.
-      (jit::tracer::isTracing() && THPVariable_Check(item.ptr())));
+        jit::tracer::isTracing() &&
+        THPVariable_Check(item.ptr()) &&
+        THPVariable_Unpack(item.ptr()).sizes() == c10::IntArrayRef{});
```

Comment:
The slow test that was failing was running tracing on the `TensorList` overload, but because of this check it was actually calling the `IntList` overload and failing.  This fixes the ambiguity in this particular test. A better solution would be if Tensors representing ints being traced were marked with a bitfield similar to `is_wrapped_number_` for wrapped scalar values. This works for now though, since you would never want to pass a scalar tensor as the `bins` argument for `histogramdd` and it seems to be the only function running into this issue.